lib/terminator.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class Terminator
+  def initialize(user_interface, database)
+    @user_interface = user_interface
+    @database = database
+  end
+
+  def run
+    loop do
+      if database.database_empty?
+        user_interface.display_no_contacts_message
+        user_interface.continue
+        return
+      end
```

Comment:
To fix the Code Climate issue you could extract the body of this if into a method, for example:

```ruby
return handle_empty_database if database.database_empty?

# blah...
```